compiler/fir/raw-fir/light-tree2fir/src/org/jetbrains/kotlin/fir/lightTree/LightTree2Fir.kt

```diff
@@ -23,6 +23,7 @@ import java.nio.file.Path
 
 class LightTree2Fir(
     val session: FirSession,
+    val headerCompilationMode: Boolean,
```

Comment:
I think this flag is not needed as you can access it via `session.languageVersionSettings.getFlag(...)`. As an example you can see how it's done for `lenientMode`. I don't see it's a good idea adding any flag we need as an argument everywhere.